packages/logic-utils/api.d.ts

```diff
@@ -128,3 +129,48 @@ export function getUI5NodeFromXMLElementNamespace(
   isDefault: boolean;
   isXmlnsDefined: boolean;
 };
+
+/**
+ * Get the deprecated message. The returned string contains jsdoc tags.
+ *
+ * @param opts.title - The opening sentance for the deprecation message. It should not contain a dot. "Deprecated" by default.
+ */
+export function getDeprecationMessage(opts: {
+  title?: string;
+  since: string | undefined;
+  text: string | undefined;
+}): string;
+
+/**
+ * Get a snippet (first line) of the deprecated documentation without jsdoc tags.
+ *
+ * @param title - The opening sentance for the deprecation message. It should not contain a dot. "Deprecated" by default.
+ * @param deprecatedInfo
+ * @param model
+ */
+export function getDeprecationPlainTextSnippet(
```

Comment:
why is `getDeprecationMessage` using **config object pattern** with optional property.
while getDeprecationPlainTextSnippet is using **ordered args** with property which may have an undefined value?

I think we should be consistent and also that with 3 or more args config object pattern should be the default for **PUBLIC** apis.

Reply:
Otherwise likely more breaking changes would occure.

Reply:
Ok. It seemed more important in `getDeprecationMessage` because that function receives 3 string arguments and here the arguments have different types so you couldn't accidentally send  the wrong argument as long as you have type checks.

Reply:
Done

Reply:
>  types so you couldn't accidentally send the wrong argument as long as you have type checks.

It is more about the ability to change in the future (e.g make some prop optional, have multiple optional props) without breaking changes

Reply:
Well, it's not exactly public API that is supposed to be used outside this mono-repo, this package is more for shared utilities between packages in the mono-repo.